### PR TITLE
[Syntax] Make FunctionTypeSyntax use a FunctionParameterList

### PR DESCRIPTION
**Pull request by harlanhaskins**
This opens up support for `inout` and labels in `FunctionTypeSyntax`, which was previously using
`TupleTypeElementList`s.

This also adds a round-trip test for function types.

Resolves [SR-4313](https://bugs.swift.org/browse/SR-4313).

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

File: unittests/Syntax/TypeSyntaxTests.cpp [hunk 1]
```diff
@@ -521,10 +533,12 @@ TEST(TypeSyntaxTests, FunctionTypeMakeAPIs) {
     auto yArg = SyntaxFactory::makeTupleTypeElement(y, Colon, Int);
```
Comment (bitjammer) [inline unittests/Syntax/TypeSyntaxTests.cpp]: Can probably nuke this x/y stuff now.

**Review by bitjammer — APPROVED** [review]
Superb.

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test and merge

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Pull request merged**